src/write.c

```diff
@@ -2672,31 +2797,37 @@ static avifResult avifEncoderWriteMiniBox(avifEncoder * encoder, avifRWStream *
     if (s->numUsedBitsInPartialByte != 0) {
         AVIF_CHECKRES(avifRWStreamWriteBits(s, 0, 8 - s->numUsedBitsInPartialByte));
     }
-    const size_t headerSize = avifRWStreamOffset(s);
+    const size_t headerBytes = avifRWStreamOffset(s);
 
     // Chunks
     if (hasAlpha && alphaData->size != 0 && codecConfigSize != 0) {
         AVIF_CHECKRES(writeCodecConfig(s, &alphaItem->av1C)); // unsigned int(8) alpha_item_codec_config[alpha_item_codec_config_size];
     }
-    // if (hdr_flag && gainmap_flag && gainmap_item_codec_config_size > 0)
-    //     unsigned int(8) gainmap_item_codec_config[gainmap_item_codec_config_size];
+    if (hasHdr && hasGainmap && codecConfigSize != 0) {
+        AVIF_CHECKRES(writeCodecConfig(s, &gainmapItem->av1C)); // unsigned int(8) gainmap_item_codec_config[gainmap_item_codec_config_size];
+    }
     if (codecConfigSize > 0) {
         AVIF_CHECKRES(writeCodecConfig(s, &colorItem->av1C)); // unsigned int(8) main_item_codec_config[main_item_codec_config_size];
     }
 
     if (hasIcc) {
         AVIF_CHECKRES(avifRWStreamWrite(s, image->icc.data, image->icc.size)); // unsigned int(8) icc_data[icc_data_size_minus1 + 1];
     }
-    // if (hdr_flag && gainmap_flag && tmap_icc_flag)
-    //     unsigned int(8) tmap_icc_data[tmap_icc_data_size_minus1 + 1];
-    // if (hdr_flag && gainmap_flag && gainmap_metadata_size > 0)
-    //     unsigned int(8) gainmap_metadata[gainmap_metadata_size];
+#if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
+    if (hasHdr && hasGainmap && tmapIccSize != 0) {
+        AVIF_CHECKRES(avifRWStreamWrite(s, encoder->data->altImageMetadata->icc.data, encoder->data->altImageMetadata->icc.size)); // unsigned int(8) tmap_icc_data[tmap_icc_data_size_minus1 + 1];
```

Comment:
If `tmapIccSiz` is nonzero, I think it is equal to `encoder->data->altImageMetadata->icc.size`. It is confusing to see `tmapIccSize` and `encoder->data->altImageMetadata->icc.size` both used here. Maybe it is better to replace `tmapIccSize` with a `tmapHasIcc` or `tmapIccFlag` boolean.

Reply:
> replace `tmapIccSize` with a `tmapHasIcc`

We should do that once `AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP` is removed. Currently it would make the `expectedChunkBytes` definition hard to read.